backend/src/main/java/com/huellapositiva/domain/actions/ManageNewsletterExcelAction.java

```diff
@@ -0,0 +1,69 @@
+package com.huellapositiva.domain.actions;
+
+import com.huellapositiva.domain.model.valueobjects.EmailAddress;
+import com.huellapositiva.domain.service.EmailCommunicationService;
+import com.huellapositiva.domain.service.RemoteStorageService;
+import com.huellapositiva.infrastructure.orm.entities.JpaVolunteer;
+import com.huellapositiva.infrastructure.orm.repository.JpaVolunteerRepository;
+import lombok.RequiredArgsConstructor;
+import org.apache.poi.xssf.usermodel.XSSFSheet;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.*;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class ManageNewsletterExcelAction {
+
+    @Autowired
+    private final JpaVolunteerRepository jpaVolunteerRepository;
+
+    private final RemoteStorageService remoteStorageService;
+
+    private final EmailCommunicationService communicationService;
+
+    private static final String ROOT = "./newsletterEmails.xlsx";
+
+    public void execute(String email) throws IOException {
+        List<JpaVolunteer> subscribedVolunteers = jpaVolunteerRepository.findSubscribedVolunteers();
+        if (subscribedVolunteers.isEmpty()){
+            communicationService.sendEmptyNewsletter(EmailAddress.from(email));
+        } else {
```

Comment:
Might to send an Exception and not an email?

Reply:
Solved